substrate/primitives/runtime/src/traits/transaction_extension/dispatch_transaction.rs

```diff
@@ -0,0 +1,144 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The [DispatchTransaction] trait.
+
+use super::*;
+
+/// Single-function utility trait with a blanket impl over [TransactionExtension] in order to
+/// provide transaction dispatching functionality. We avoid implementing this directly on the trait
+/// since we never want it to be overriden by the trait implementation.
+pub trait DispatchTransaction<Call: Dispatchable> {
```

Comment:
This trait needs to be in primitives?

I wanted to use the dispatch context https://paritytech.github.io/polkadot-sdk/master/frame_support/dispatch_context/index.html
inside the transaction extensions. But transaction extensions are not inside the dispatch context.

IMO as we move more logic in the transaction extensions providing the dispatch context would also be great. But for this we should move dispatch context code and this code in the same crate (primitives or frame).

But we can do in follow-up.

Reply:
This trait is technically just a helper trait, the main traits that are used within substrate are actually `Checkable` and `Applyable`, which happen to make use of this helper trait. But yes, we could move it later to be more convenient.